File: frontend/src/models/assist_accounting.rs
//! 辅助核算模型
//!
//! 辅助核算相关的数据结构

use serde::{Deserialize, Serialize};
use serde_json;

/// 辅助核算维度
#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct AssistDimension {
    pub id: i32,
    pub dimension_code: String,
    pub dimension_name: String,
    pub description: Option<String>,
    pub is_active: bool,
    pub sort_order: i32,
}

/// 辅助核算记录
#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct AssistRecord {
    pub id: i32,
    pub business_type: String,
    pub business_no: String,
    pub business_id: i32,
    pub account_subject_id: i32,
    pub debit_amount: serde_json::Value,
    pub credit_amount: serde_json::Value,
    pub five_dimension_id: String,
    pub product_id: i32,
    pub batch_no: String,
    pub color_no: String,
    pub dye_lot_no: Option<String>,
    pub grade: String,
    pub warehouse_id: i32,
    pub quantity_meters: serde_json::Value,
    pub quantity_kg: serde_json::Value,
    pub workshop_id: Option<i32>,
    pub customer_id: Option<i32>,
    pub supplier_id: Option<i32>,
    pub remarks: Option<String>,
    pub created_at: String,
}

/// 辅助核算汇总
#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct AssistSummary {
    pub id: i32,
    pub accounting_period: String,
    pub dimension_code: String,
    pub dimension_value_id: i32,
    pub dimension_value_name: String,
    pub account_subject_id: i32,
    pub total_debit: serde_json::Value,
    pub total_credit: serde_json::Value,
    pub total_quantity_meters: serde_json::Value,
    pub total_quantity_kg: serde_json::Value,
    pub record_count: i64,
}

/// 辅助核算记录列表响应
#[derive(Debug, Clone, Deserialize)]
pub struct AssistRecordListResponse {
    pub records: Vec<AssistRecord>,
    #[allow(dead_code)]
    pub total: u64,
    #[allow(dead_code)]
    pub page: u64,
    #[allow(dead_code)]
    pub page_size: u64,
}

/// 辅助核算查询参数
#[derive(Debug, Clone, Serialize)]
pub struct AssistRecordQueryParams {
    pub accounting_period: Option<String>,
    pub dimension_code: Option<String>,
    pub business_type: Option<String>,
    pub warehouse_id: Option<i32>,
    pub page: Option<u64>,
    pub page_size: Option<u64>,
}

/// 辅助核算汇总查询参数
#[derive(Debug, Clone, Serialize)]
pub struct AssistSummaryQueryParams {
    pub accounting_period: String,
    pub dimension_code: Option<String>,
}

/// 业务单查询参数
#[allow(dead_code)]
#[derive(Debug, Clone, Serialize)]
pub struct BusinessQueryParams {
    pub business_type: String,
    pub business_no: String,
}

File: frontend/src/api/assistAccounting.ts
import { request } from './request'

export interface AssistDimensionResponse {
  id: number
  dimension_code: string
  dimension_name: string
  description?: string
  is_active: boolean
  sort_order: number
}

export interface AssistRecordResponse {
  id: number
  business_type: string
  business_no: string
  business_id: number
  account_subject_id: number
  debit_amount: number
  credit_amount: number
  five_dimension_id: string
  product_id: number
  batch_no: string
  color_no: string
  dye_lot_no?: string
  grade: string
  warehouse_id: number
  quantity_meters: number
  quantity_kg: number
  workshop_id?: number
  customer_id?: number
  supplier_id?: number
  remarks?: string
  created_at: string
}

export interface AssistSummaryResponse {
  id: number
  accounting_period: string
  dimension_code: string
  dimension_value_id: number
  dimension_value_name: string
  account_subject_id: number
  total_debit: number
  total_credit: number
  total_quantity_meters: number
  total_quantity_kg: number
  record_count: number
}

export interface AssistRecordListResponse {
  records: AssistRecordResponse[]
  total: number
  page: number
  page_size: number
}

export interface AssistRecordQueryParams {
  accounting_period?: string
  dimension_code?: string
  business_type?: string
  warehouse_id?: number
  page?: number
  page_size?: number
}

export interface BusinessQueryParams {
  business_type: string
  business_no: string
}

export interface AssistSummaryQueryParams {
  accounting_period: string
  dimension_code?: string
}

export function listAssistDimensions() {
  return request.get('/assist-accounting/dimensions')
}

export function queryAssistRecords(params?: AssistRecordQueryParams) {
  return request.get('/assist-accounting/records', { params })
}

export function getAssistRecordsByBusiness(params: BusinessQueryParams) {
  return request.get('/assist-accounting/records/by-business', { params })
}

export function getAssistRecordsByFiveDimension(fiveDimensionId: string) {
  return request.get(`/assist-accounting/records/five-dimension/${fiveDimensionId}`)
}

export function getAssistSummary(params: AssistSummaryQueryParams) {
  return request.get('/assist-accounting/summary', { params })
}